stores/sql/main.go

```diff
@@ -2527,81 +2576,33 @@ func Object(ctx context.Context, tx Tx, bucket, key string) (api.Object, error)
 
 	// fetch slab slices
 	rows, err = tx.Query(ctx, `
-		SELECT sla.db_buffered_slab_id IS NOT NULL, sli.object_index, sli.offset, sli.length, sla.health, sla.key, sla.min_shards, COALESCE(sec.slab_index, 0), COALESCE(sec.root, ?), COALESCE(c.fcid, ?), COALESCE(c.host_key, ?)
+		SELECT sla.id, sla.health, sla.key, sla.min_shards, sli.offset, sli.length
 		FROM slices sli
 		INNER JOIN slabs sla ON sli.db_slab_id = sla.id
-		LEFT JOIN sectors sec ON sec.db_slab_id = sla.id
-		LEFT JOIN contract_sectors csec ON csec.db_sector_id = sec.id
-		LEFT JOIN contracts c ON c.id = csec.db_contract_id
 		WHERE sli.db_object_id = ?
-		ORDER BY sli.object_index ASC, sec.slab_index ASC
-	`, Hash256{}, FileContractID{}, PublicKey{}, objID)
+		ORDER BY sli.object_index ASC
+	`, objID)
 	if err != nil {
 		return api.Object{}, fmt.Errorf("failed to fetch slabs: %w", err)
 	}
 	defer rows.Close()
 
 	slabSlices := object.SlabSlices{}
-	var current *object.SlabSlice
-	var currObjIdx, currSlaIdx int64
 	for rows.Next() {
-		var bufferedSlab bool
-		var objectIndex int64
-		var slabIndex int64
+		var id int64
```

Comment:
does it make sense to collect all slab ids first?

Reply:
Probably not